torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -187,6 +187,18 @@ class CPUOffload:
     offload_params: bool = False
 
 
+@dataclass
+class ForwardPrefetch:
+    """
+    Forward full parameters prefetch config.
+    prefetch_full_params: Initialize all-gather to rebuild full parameters of
+                          the layer right before the forward pass computation
+                          of the previous layer. This approach enables communication
+                          and computation overlap in the forward pass.
```

Comment:
mention it might (will) result in memory increase?

Reply:
Thanks! Comments are added